lib/rubocop/cop/lint/unneeded_disable.rb

```diff
@@ -24,6 +24,9 @@ def check(offenses, cop_disabled_line_ranges, comments)
               comment = comments.find { |c| c.loc.line == line_range.begin }
               unneeded_cop = find_unneeded(comment, offenses, cop, cop_offenses,
                                            line_range)
+
+              return nil if unneeded_cop == COP_NAME
```

Comment:
Good catch. I wasn't thinking about that situation. This will complicate things slightly, but I am sure that I will find a solution. I am thinking that I may need to find all of the `Lint/UnneededDisable` comment ranges before any of the other cops, and then use that to compare the ranges.